shippers/es.py

```diff
@@ -142,3 +147,40 @@ def flush(self) -> Any:
             self._log_outcome(success=success, failed=failed)
 
         self._bulk_actions = []
+
+    def discover_dataset(self, event: Dict[str, Any]) -> None:
+        if self._dataset == "":
+            body: str = event["Records"][0]["body"]
+            json_body: Dict[str, Any] = json.loads(body)
+            s3_object_key: str = ""
+
+            if "Records" in json_body and len(json_body["Records"]) > 0:
+                if "s3" in json_body["Records"][0]:
+                    s3_object_key = json_body["Records"][0]["s3"]["object"]["key"]
+            else:
+                raise KeyError("Invalid event structure")
```

Comment:
please, do not `raise` here and fallback to `generic`

Reply:
here I was checking for the records, but removed it